sasmodels/model_test.py

```diff
@@ -213,18 +217,32 @@ def run_all(self):
                 # test vector form
                 ({}, [0.001, 0.01, 0.1], [None]*3),
                 ({}, [(0.1, 0.1)]*2, [None]*2),
-                # test that ER/VR will run if they exist
-                ({}, 'ER', None),
-                ({}, 'VR', None),
+                # test that Fq will run, and return R_eff, V, V_r
+                ({}, 0.1, None, None, None, None, None),
                 ]
             tests = smoke_tests
             #tests = []
             if self.info.tests is not None:
                 tests += self.info.tests
+            S_tests = [test for test in tests if '@S' in test[0]]
+            P_tests = [test for test in tests if '@S' not in test[0]]
```

Comment:
Alternatively, to avoid double iteration:
S_tests, P_tests = [], []
for t in tests:
    (S_tests if '@S' in t[0] else P_tests).append(t)

Reply:
Yes I found the following on stackoverflow when I first wrote this
https://stackoverflow.com/questions/949098/python-split-a-list-based-on-a-condition

I don't like any of the options:
```
# option 1
S_tests = [test for test in tests if "@S" in test[0]]
P_tests = [test for test in tests if "@S" not in test[0]]

# option 2
S_tests, P_tests = [], []
for t in tests:
         (S_tests if '@S' in t[0] else P_tests).append(t)

# option 3
S_tests, P_tests = [], []
for t in tests:
         (S_tests, P_tests)['@S' in t[0]].append(t)

# option 4
from .util import split_on_condition
...
S_tests, P_tests = split_on_condition(tests, lambda test: "@S" in test)

# with the following in util.py
def split_on_condition(seq, condition):
    a, b = [], []
    for item in seq:
        (a if condition(item) else b).append(item)
    return a, b
```

Since the structures we are working with are tiny we can ignore performance (1.6 us vs 3.2 us on this machine) and choose whatever is easiest to understand.  I'm guessing it is the first one.

Reply:
If self.info.tests is indeed a small list (it likely is), then the discussion is moot - your double iteration assignment is the most explicit expression.